rolling-update-kubernetes/src/test/scala/akka/rollingupdate/kubernetes/AppVersionRevisionSpec.scala

```diff
@@ -0,0 +1,187 @@
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.{ aResponse, get, stubFor, urlEqualTo }
+import com.github.tomakehurst.wiremock.client.{ MappingBuilder, ResponseDefinitionBuilder, WireMock }
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
+import com.github.tomakehurst.wiremock.matching.EqualToPattern
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.{ Millis, Seconds, Span }
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.scalatest.{ BeforeAndAfterAll, BeforeAndAfterEach }
+
+import scala.concurrent.duration._
+
+object AppVersionRevisionSpec {
+  val config = ConfigFactory.parseString("""
+      akka.loggers = ["akka.testkit.TestEventListener"]
+      akka.actor.provider = cluster
+
+      akka.remote.artery.canonical.port = 0
+      akka.remote.artery.canonical.hostname = 127.0.0.1
+
+      akka.cluster.jmx.multi-mbeans-in-same-jvm = on
+      akka.coordinated-shutdown.terminate-actor-system = off
+      akka.coordinated-shutdown.run-by-actor-system-terminate = off
+      akka.test.filter-leeway = 10s
+    """)
+}
+
+class AppVersionRevisionSpec
+    extends TestKit(
+      ActorSystem(
+        "AppVersionRevisionSpec",
+        AppVersionRevisionSpec.config
+      ))
+    with ImplicitSender
+    with AnyWordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with Eventually {
+
+  private val wireMockServer = new WireMockServer(wireMockConfig().port(0))
+  wireMockServer.start()
+  WireMock.configureFor(wireMockServer.port())
+
+  // for wiremock to provide json
+  val mapper = new ObjectMapper()
+
+  private val namespace = "namespace-test"
+  private val podName1 = "pod-test-1"
+
+  private def settings(podName: String) = {
+    new KubernetesSettings(
+      apiCaPath = "",
+      apiTokenPath = "",
+      apiServiceHost = "localhost",
+      apiServicePort = wireMockServer.port(),
+      namespace = Some(namespace),
+      namespacePath = "",
+      podName = podName,
+      secure = false,
+      apiServiceRequestTimeout = 2.seconds,
+      customResourceSettings = new CustomResourceSettings(enabled = false, crName = None, 60.seconds)
+    )
+  }
+
+  private val kubernetesApi =
+    new KubernetesApiImpl(
+      system,
+      settings(podName1),
+      namespace,
+      apiToken = "apiToken",
+      clientHttpsConnectionContext = None)
+
+  override implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(timeout = Span(5, Seconds), interval = Span(100, Millis))
+
+  override protected def afterAll(): Unit = super.shutdown()
+
+  override protected def beforeEach(): Unit = {
+    wireMockServer.resetAll()
+  }
+
+  def expectLogInfo[T](pattern: String = null)(block: => T): T =
```

Comment:
This and a few other bits were taken from an existing spec. Should we move them to a common place?

Reply:
I don't think tests have to be completely DRY. This specific `def expectLogInfo` doesn't have much value at all in my opinion, compared to just inlining 
```
EventFilter.info(pattern = pattern, occurrences = 1).intercept {
  ...
}